### PR TITLE
Add support for keyword lists

### PR DESCRIPTION
**Pull request by matthijsqd**
For one of our projects we needed support for keyword lists, so that the key-value pairs would be encoded in the correct order in the YAML output. Currently trying to convert a keyword list will result in an exception. This PR adds a pattern match to `encode_as_io_list` so that keyword lists are encoded in the same way as maps.

---

#### Requirements

- [x] Entry in CHANGELOG.md was created
- [ ] ~Link to documentation on https://yaml.org/ is provided in the PR description~
- [x] Functionality is covered by newly created tests

### CODE REVIEW

File: lib/ymlr/encoder_test.exs [hunk 1]
```diff
@@ -168,6 +168,18 @@ defmodule Ymlr.EncoderTest do
...
 
+    test "keyword lists" do
+      assert MUT.to_s!([a: 1]) == "a: 1"
+      assert MUT.to_s!([a: 1, b: 2]) == "a: 1\nb: 2"
+      assert MUT.to_s!([a: nil]) == "a:"
+    end
+    
+    test "invalid keyword key" do
+      assert_raise ArgumentError, fn ->
+        MUT.to_s!([{"a", 1}])
```
Comment (jlgeering) [inline lib/ymlr/encoder_test.exs:179]: don't think I agree with this: it is not a valid keyword list, but it is a valid list with 1 tuple element ...

so, still not supported but the name of the test should probably be "list of tuples - tuples not supported"

=> and we should probably also add a "tuples not supported" test to make it explicit

Comment (jlgeering) [reply lib/ymlr/encoder_test.exs:179]: new tests done in e5460045b68399b16d46271bcd6494776850ccfa

**Comment by jlgeering** [comment]
actually, it gets worse (more complicated): how should `[{:a, 1}, 2]` or `[{:a, 1}, :b]` or `[:a, {:b, 2}]` be encoded? or should it throw an error?

with this change, `Ymlr.Encoder.to_s!([:a, {:b, 2}])` returns `"- a\nb: 2"` and that is not valid yaml

**Comment by jlgeering** [comment]
we might have to use https://hexdocs.pm/elixir/1.12/Keyword.html#keyword?/1 in` encode_as_io_list(data, level) when is_list(data)`

**Comment by jlgeering** [comment]
and ... we should add a test to make it explicit what happens with duplicate keys `[{:a, 1}, {:a, 2}]`

probably pick the first one, as `Keyword.get([a: 1, a: 2], :a)` returns 1

`assert MUT.to_s!([a: 1, a: 2]) == "a: 1"`

**Comment by jlgeering** [comment]
@matthijsqd do you mind rebasing on main and taking another stab at it?

**Comment by mruoss** [comment]
> probably pick the first one, as `Keyword.get([a: 1, a: 2], :a)` returns 1

Probably. The alternative would be to use `Keyword.values()` and make the value a list.

**Comment by matthijsqd** [comment]
> @matthijsqd do you mind rebasing on main and taking another stab at it?

Fair points, I'll give it another shot

**Comment by mruoss** [comment]
With #118 tuples will be encoded as lists. So there's not gonna be a support for keyword lists being encoded like maps. 

On that branch, this happens:

```elixir
Ymlr.document!(a: "a", b: "b") 
"""
---
- - a
  - a
- - b
  - b
"""
```

Going to close this if no objections.
